lib/broadway_rabbitmq/amqp_client.ex

```diff
@@ -200,15 +200,39 @@ defmodule BroadwayRabbitMQ.AmqpClient do
 
   @impl true
   def setup_channel(config) do
-    with {name, config} <- Map.pop(config, :name, :undefined),
-         {:ok, conn} <- Connection.open(config.connection, name),
-         {:ok, channel} <- Channel.open(conn),
-         :ok <- call_after_connect(config, channel),
-         :ok <- Basic.qos(channel, config.qos),
-         {:ok, queue} <- maybe_declare_queue(channel, config.queue, config.declare_opts),
-         :ok <- maybe_bind_queue(channel, queue, config.bindings) do
-      {:ok, channel}
+    {name, config} = Map.pop(config, :name, :undefined)
+
+    case Connection.open(config.connection, name) do
+      {:ok, conn} ->
+        true = Process.link(conn.pid)
```

Comment:
If we only monitor, and our process crashes, the connection is left open 😮

Reply:
But if the connection crashes, now it brings down the producer and we can't reconnect. Is that what we want?

Reply:
Answer: we are trapping exits, so this line is fine.